lib/Doctrine/ORM/Id/AssignedGenerator.php

```diff
@@ -23,7 +21,7 @@ class AssignedGenerator extends AbstractIdGenerator
      */
     public function generate(EntityManager $em, $entity)
     {
-        $class      = $em->getClassMetadata(get_class($entity));
+        $class      = $em->getClassMetadata($entity::class);
```

Comment:
Can `$entity` be `null`? If not, then `$entity::class` shouldn't be used.

Reply:
Same as above, `get_class()` fails on `null` as well.

`AbstractIdGenerator::generate()` declares the parameter as nullable, but that doesn't feel right. Maybe we should evaluate tighten the parameter type here, but I'd consider this out of scope for this PR.